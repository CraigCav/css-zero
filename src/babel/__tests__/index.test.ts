import path from 'path';
import { transformFileSync } from '@babel/core';
import dce from 'babel-plugin-remove-unused-vars';
import plugin from '../';

expect.addSnapshotSerializer({
  test: value => value && typeof value.cssZero === 'object',
  print: ({ cssZero: { rules } }) =>
    Object.entries(rules)
      .map(([key, value]: any) => `${key} {${value.cssText}}`)
      .join('\n'),
});

const transpile = (file: string) =>
  transformFileSync(path.resolve(__dirname, file), {
    plugins: [plugin, dce],
  });

it.each([['simple.jsx'], ['combining.jsx'], ['merging.jsx']])('%s', file => {
  const { code, metadata } = transpile(`./fixtures/${file}`)!;
  expect(code).toMatchSnapshot();
  expect(metadata).toMatchSnapshot();
});
