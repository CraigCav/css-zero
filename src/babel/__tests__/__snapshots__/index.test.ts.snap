// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combining.jsx 1`] = `"export const Component = () => <div className={\`x1dqz7z3 x1e4w2a9\`} />;"`;

exports[`combining.jsx 2`] = `
.x1dqz7z3 {color:red}
.x1e4w2a9 {font-size:16px}
`;

exports[`conditional.jsx 1`] = `
"export const LogicalAndExpression = props => <div className={\`\${props.isBlue ? \`x1vong5g\` : \`x1dqz7z3\`} x1e4w2a9\`} />;
export const LogicalExpressionDeterministic = () => <div className={\`x1vong5g\`} />;
export const IgnoreFalsey = () => <div className={\`\`} />;
export const SimpleTernaryExpression = () => <div className={\`x1dqz7z3 x1e4w2a9\`} />;"
`;

exports[`conditional.jsx 2`] = `
.x1vong5g {color:blue}
.x1dqz7z3 {color:red}
.x1e4w2a9 {font-size:16px}
`;

exports[`merging.jsx 1`] = `"export const Component = () => <div className={\`xp5623v\`} />;"`;

exports[`merging.jsx 2`] = `.xp5623v {color:green}`;

exports[`simple.jsx 1`] = `"export const Component = () => <div className={\`x1e4w2a9\`} />;"`;

exports[`simple.jsx 2`] = `.x1e4w2a9 {font-size:16px}`;
